tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -4744,67 +4707,45 @@ Status FragmentMetadata::write_tile_null_counts(unsigned idx, Buffer* buff) {
     uint64_t num, const EncryptionKey& encryption_key, uint64_t* nbytes) {
   Status st;
   Buffer buff;
 
-  // Store all attributes.
-  for (unsigned int i = 0; i < num; ++i) {
-    // Store min.
-    uint64_t min_size = fragment_mins_[i].size();
-    st = buff.write(&min_size, sizeof(uint64_t));
-    if (!st.ok()) {
-      return LOG_STATUS(
-          Status_FragmentMetadataError("Cannot serialize fragment metadata; "
-                                       "Writing fragment min size failed"));
-    }
+  auto serialize_data = [&](Serializer& serializer) {
+    // Store all attributes.
+    for (unsigned int i = 0; i < num; ++i) {
+      // Store min.
+      uint64_t min_size = fragment_mins_[i].size();
+      serializer.write<uint64_t>(min_size);
 
-    st = buff.write(fragment_mins_[i].data(), min_size);
-    if (!st.ok()) {
-      return LOG_STATUS(
-          Status_FragmentMetadataError("Cannot serialize fragment metadata; "
-                                       "Writing fragment min failed"));
-    }
+      serializer.write(fragment_mins_[i].data(), min_size);
 
-    // Store max.
-    uint64_t max_size = fragment_maxs_[i].size();
-    st = buff.write(&max_size, sizeof(uint64_t));
-    if (!st.ok()) {
-      return LOG_STATUS(
-          Status_FragmentMetadataError("Cannot serialize fragment metadata; "
-                                       "Writing fragment max size failed"));
-    }
+      // Store max.
+      uint64_t max_size = fragment_maxs_[i].size();
+      serializer.write<uint64_t>(max_size);
 
-    st = buff.write(fragment_maxs_[i].data(), max_size);
-    if (!st.ok()) {
-      return LOG_STATUS(
-          Status_FragmentMetadataError("Cannot serialize fragment metadata; "
-                                       "Writing fragment max failed"));
-    }
+      serializer.write(fragment_maxs_[i].data(), max_size);
 
-    // Store sum.
-    st = buff.write(&fragment_sums_[i], sizeof(uint64_t));
-    if (!st.ok()) {
-      return LOG_STATUS(
-          Status_FragmentMetadataError("Cannot serialize fragment metadata; "
-                                       "Writing fragment sum failed"));
-    }
+      // Store sum.
+      serializer.write<uint64_t>(fragment_sums_[i]);
 
-    // Store null count.
-    st = buff.write(&fragment_null_counts_[i], sizeof(uint64_t));
-    if (!st.ok()) {
-      return LOG_STATUS(
-          Status_FragmentMetadataError("Cannot serialize fragment metadata; "
-                                       "Writing fragment null count failed"));
+      // Store null count.
+      serializer.write<uint64_t>(fragment_null_counts_[i]);
     }
-  }
+  };
 
-  RETURN_NOT_OK(write_generic_tile_to_file(encryption_key, buff, nbytes));
+  SizeComputationSerializer size_computation_serializer;
+  serialize_data(size_computation_serializer);
 
-  storage_manager_->stats()->add_counter("write_null_counts_size", *nbytes);
+  Tile tile{Tile::from_generic(size_computation_serializer.size())};
```

Comment:
Could you fix the other places in the code to use from_generic? You can search for:
Serializer serializer(tile.data(), tile.size());

Reply:
There is one more in storage_manager.cc (in StorageManager::store_array_schema) 🙏